WordPress/gutenberg
# Editor: Move PluginPostPublishPanel and PluginPrePublishPanel to editor package

## What?
Related: https://github.com/WordPress/gutenberg/issues/52632, https://github.com/WordPress/gutenberg/issues/59878
Extracted from: https://github.com/WordPress/gutenberg/pull/60289

This PR  moves PluginPostPublishPanel and PluginPrePublishPanel to editor package. This is a prerequisite for unifying the publishing flow between post and site editors.


## Testing Instructions
1. Observe that previous panel usage from `edit-post` packages works as before
2. Test by rendering something in the slots from `editor` package.

Example code:
```
import {
	PluginPostPublishPanel,
	PluginPrePublishPanel,
} from '@wordpress/editor';
const MyPluginPostPublishPanel = () => {
	return (
		<>
			<PluginPostPublishPanel
				className="my-plugin-post-publish-panel"
				title={ __( 'My post panel title' ) }
				initialOpen
			>
				{ __( 'My post publish panel content' ) }
			</PluginPostPublishPanel>
			<PluginPrePublishPanel
				className="my-plugin-post-publish-panel"
				title={ __( 'My pre panel title' ) }
				initialOpen
			>
				{ __( 'My pre publish panel content' ) }
			</PluginPrePublishPanel>
		</>
	);
};
```

### Notes

The editor package doesn't auto generate docs and will explore this separately. I'll have a PR soon for that.
